src/bindings/jna/libelektra/src/test/java/org/libelektra/KeySetTest.java

```diff
@@ -814,4 +809,500 @@ public void test_keySetTailSetShouldPass() {
     assertTrue(keySet.contains(key5));
     assertTrue(tailSet.contains(key6));
   }
+
+  @Test
+  public void test_keySetLower_shouldPass() {
+    var keySet = KeySet.create(5, key2, key3, key4, key5);
+
+    assertNull(keySet.lower(key));
+    assertNull(keySet.lower(key2));
+    assertEquals(key2, keySet.lower(key3));
+    assertEquals(key3, keySet.lower(key4));
+    assertEquals(key4, keySet.lower(key5));
+    assertNull(keySet.lower(key6));
+  }
+
+  @Test
+  public void test_keySetFloor_shouldPass() {
+    var keySet = KeySet.create(5, key2, key3, key4, key5);
+
+    assertNull(keySet.floor(key));
+    assertEquals(key2, keySet.floor(key2));
+    assertEquals(key3, keySet.floor(key3));
+    assertEquals(key4, keySet.floor(key4));
+    assertEquals(key5, keySet.floor(key5));
+    assertNull(keySet.floor(key6));
```

Comment:
This assertion is also wrong. `set.floor(x)` and `set.lower(x)` must return the same element, if `set.contains(x) == false`. Only if `set.contains(x) == true` will `set.floor(x) == x` and `set.lower(x)` will be different.

Also no checks with missing keys in the middle.